Diagonal walls and secret doors

ErhardSteinhauer opened the pull request:
## About the PR
New entities:
- Diagonal walls: wood, reinforced and uranium.
- Secret doors (can be constructed): wood, reinforced, shuttle, uranium.

## Why / Balance
More structures for mappers and smugglers.

## Technical details
new .yml and .rsi

## Media
![image](https://github.com/new-frontiers-14/frontier-station-14/assets/65374927/455d93c1-465d-4138-8a12-6200abfc68b2)
Fig. 1 - Structures (secret doors aren't present since they look like walls anyways, notice boards were moved to separate PR).
- [X] I have added screenshots/videos to this PR showcasing its changes ingame.

**Changelog**
:cl: erhardsteinhauer
- add: Added buildable entities: secret doors (wood, reinforced, shuttle, uranium). And some diagonal walls for mappers.

FoxxoTrystan commented on Resources/Prototypes/_NF/Entities/Structures/Walls/diagonal_walls.yml:
WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

ErhardSteinhauer replied:
> WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

It is parent for solid diagonal wall in the base game. I can explicitly specify the reflective component as 0.

FoxxoTrystan commented on Resources/Prototypes/_NF/Entities/Structures/Walls/grille_diagonal.yml:
Same issue:

WallShuttleDiagonal should not be a parent of Diagonals walls, this parent got reflective

ErhardSteinhauer commented on Resources/Prototypes/_NF/Entities/Structures/Doors/secret_door.yml:
Sure

ErhardSteinhauer commented on Resources/Prototypes/_NF/Entities/Structures/Windows/window.yml:
My bad thought that it from my previous iteration of this branch

ErhardSteinhauer commented:
Changed a few things around:
- Kegs moved to separate PR.
- Notice board moved to separate PR.
- Changed parent on diagonal windows from `BaseItem` to `Window`.
- Changed parent on diagonal walls from `WallShuttleDiagonal` to `WallSolidDiagonal`.
- Removed resistances from secret doors.
- Restored `windows.yml`.

ErhardSteinhauer commented:
Made so diagonal windows inherit from their respective base variations: eg., diagonal plasma window inherits from plasma window etc.

dvir001 commented:
I made multi parents changed to walls, all seem fine to me after testing, please confirm on your side its all good.

ErhardSteinhauer commented:
> I made multi parents changed to walls, all seem fine to me after testing, please confirm on your side its all good.

Your version works but I made some changes too and those appear to work fine on my end:
- Added new entity to use as a parent instead of WallSolidDiagonal (which inherits from WallShuttleDiagonal and has reflect component) - BaseWallDiagonal.
- Replaced WallSolidDiagonal with BaseWallDiagonal everywhere.

ErhardSteinhauer commented:
Also what's up with Map Prototype Checker throwing errors?

ErhardSteinhauer commented:
Moving diagonal windows and grille to a separate PR, leaving here here diagonal walls and secret doors.
